### ISSUE TITLE
App ops fail to change.

### ISSUE THREAD

**Issue by BloodReleasesPain** [MuntashirAkon/AppManager#522]
- [x] I know what my device, OS and App Manager versions are
- [x] I know how to take logs
- [x] I know how to reproduce the issue which may not be specific to my device

When changing app ops, the setting chosen will not stick after refreshing apps ops, and it's respective permission switch will also get stuck.

Log:

```
2021-07-21 07:41:50.834 E/ADA - App Ops: isPackageChanged called
2021-07-21 07:42:28.930 E/AppOpsService: Could not get app ops for UID 10154
java.lang.NullPointerException: Attempt to invoke virtual method 'int android.util.SparseIntArray.size()' on a null object reference
	at android.os.Parcel.readException(Parcel.java:1973)
	at android.os.Parcel.readException(Parcel.java:1913)
	at com.android.internal.app.IAppOpsService$Stub$Proxy.getUidOps(IAppOpsService.java:579)
	at io.github.muntashirakon.AppManager.appops.AppOpsService.getOpsForPackage(AppOpsService.java:67)
	at io.github.muntashirakon.AppManager.appops.AppOpsUtils.getChangedAppOps(AppOpsUtils.java:15)
	at io.github.muntashirakon.AppManager.details.AppDetailsViewModel.loadAppOps(AppDetailsViewModel.java:1125)
	at io.github.muntashirakon.AppManager.details.AppDetailsViewModel.lambda$l0EWUQ8pwSEwIOu4g2r5od7Q6Ks(Unknown Source:0)
	at io.github.muntashirakon.AppManager.details.-$$Lambda$AppDetailsViewModel$l0EWUQ8pwSEwIOu4g2r5od7Q6Ks.run(Unknown Source:2)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:458)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
	at java.lang.Thread.run(Thread.java:764)
2021-07-21 07:42:30.615 E/ADA - App Ops: isPackageChanged called
2021-07-21 07:42:30.653 E/AppOpsService: Could not get app ops for UID 10154
java.lang.NullPointerException: Attempt to invoke virtual method 'int android.util.SparseIntArray.size()' on a null object reference
	at android.os.Parcel.readException(Parcel.java:1973)
	at android.os.Parcel.readException(Parcel.java:1913)
	at com.android.internal.app.IAppOpsService$Stub$Proxy.getUidOps(IAppOpsService.java:579)
	at io.github.muntashirakon.AppManager.appops.AppOpsService.getOpsForPackage(AppOpsService.java:67)
	at io.github.muntashirakon.AppManager.appops.AppOpsUtils.getChangedAppOps(AppOpsUtils.java:15)
	at io.github.muntashirakon.AppManager.details.AppDetailsViewModel.loadAppOps(AppDetailsViewModel.java:1125)
	at io.github.muntashirakon.AppManager.details.AppDetailsViewModel.lambda$load$9$AppDetailsViewModel(AppDetailsViewModel.java:699)
	at io.github.muntashirakon.AppManager.details.-$$Lambda$AppDetailsViewModel$llrumPmzrbtvwWMKkJUJipfWAtE.run(Unknown Source:4)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:458)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
	at java.lang.Thread.run(Thread.java:764)
2021-07-21 07:42:34.854 E/ADA - Uses permissions: isPackageChanged called
2021-07-21 07:42:34.891 E/AppOpsService: Could not get app ops for UID 10154
java.lang.NullPointerException: Attempt to invoke virtual method 'int android.util.SparseIntArray.size()' on a null object reference
	at android.os.Parcel.readException(Parcel.java:1973)
	at android.os.Parcel.readException(Parcel.java:1913)
	at com.android.internal.app.IAppOpsService$Stub$Proxy.getUidOps(IAppOpsService.java:579)
	at io.github.muntashirakon.AppManager.appops.AppOpsService.getOpsForPackage(AppOpsService.java:67)
	at io.github.muntashirakon.AppManager.appops.AppOpsUtils.getChangedAppOps(AppOpsUtils.java:15)
	at io.github.muntashirakon.AppManager.details.AppDetailsViewModel.loadAppOps(AppDetailsViewModel.java:1125)
	at io.github.muntashirakon.AppManager.details.AppDetailsViewModel.lambda$load$9$AppDetailsViewModel(AppDetailsViewModel.java:699)
	at io.github.muntashirakon.AppManager.details.-$$Lambda$AppDetailsViewModel$llrumPmzrbtvwWMKkJUJipfWAtE.run(Unknown Source:4)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:458)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
	at java.lang.Thread.run(Thread.java:764)
```

 - Galaxy S7 Edge 
 - 8.0.0
 - Debug #1420
 - ADB over TCP

**Comment by MuntashirAkon** [comment]
> When changing app ops, the setting chosen will not stick after refreshing apps ops, and it's respective permission switch will also get stuck.

Could you describe it more. It's better if you can give some examples.

**Comment by BloodReleasesPain** [comment]
> > When changing app ops, the setting chosen will not stick after refreshing apps ops, and it's respective permission switch will also get stuck.
> 
> Could you describe it more. It's better if you can give some examples.

Sure. If I turn off all permissions of an app from its app info page or another app, (Not AM), then go back to AM and refresh permissions, it will still show everything is enabled.

![Screenshot_20210730-110312_Package installer.jpg]

(https://user-images.githubusercontent.com/43686736/127694061-65ef5353-4105-4bde-a054-a67b0de6875f.jpg)

I originally thought app ops only had this issue but I found out accidentally after clearing cache and data of an app I went to AM to re-enable everything and to my surprise it showed even after a refresh everything was enabled when they really weren't. So in my orginal issue it shouldn't be "App ops" as much as "Permissions", but permissions that have an entry always in both app ops and permissions like WRITE_SETTINGS will show "enabled" when it should be "default" by default of course.
